### PR TITLE
Initial implementation of navigation and redux

### PR DESCRIPTION
**Pull request by putuyoga**
This is the base foundation app for mobile for now.
Initial implementation including:
1. Navigation implementation
2. Redux implementation
3. Basic dumb Movie card component

### CODE REVIEW

File: src/appNavigator.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { StackNavigator } from 'react-navigation';
+import HomeScreen from './pages/HomeScreen';
+import MovieDetailScreen from './pages/MovieDetailScreen';
+
+const AppNavigator = StackNavigator({
+  Home: { screen: HomeScreen },
+  MovieDetail: { screen: MovieDetailScreen }
+});
+
+export default AppNavigator;
```
Comment (BAOTRAN1990) [inline src/appNavigator.js]: Is it better if we move this file to navigations folder?

File: src/appWithNavigationState.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        dispatch: this.props.dispatch,
+        state: this.props.nav,
+      })} />
+    );
+  }
+}
+
+export default connect(state => ({
+  nav: state.nav,
+}))(AppWithNavigationState);
```
Comment (BAOTRAN1990) [inline src/appWithNavigationState.js]: Is it better if we move this file to navigations folder?

File: src/pages/HomeScreen.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
+    this.state = {
+      dataSource: ds.cloneWithRows(movies),
+    };
+  }
+
+  render() {
+    const { navigate } = this.props.navigation;
+    return (
+        <ListView dataSource={this.state.dataSource} renderRow={(rowData) => <Button onPress={() => navigate('MovieDetail')} title={rowData.name} />}/>
```
Comment (BAOTRAN1990) [inline src/pages/HomeScreen.js]: It's clearer if you move renderRow function implementation to outside of our component.

File: src/pages/MovieDetailScreen.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  }
+
+  render() {
+    return (
+        <Text>This is Detail Page. Should be implement REDUX after this!</Text>
+    );
+  }
+}
+
+export default MovieDetailScreen;
```
Comment (BAOTRAN1990) [inline src/pages/MovieDetailScreen.js]: I think you should name containers folder instead of page.

File: src/pages/HomeScreen.js (hunk 3 above)
Comment (BAOTRAN1990) [reply src/pages/HomeScreen.js]: I think you should name containers folder instead of page.

**Comment by putuyoga** [comment]
I updated the app structures ( like rename the directory `pages` to `containers`) @BAOTRAN1990

**Comment by BAOTRAN1990** [comment]
Thanks Yoga. Merged!

**Pull request merged**